src/main/scala/org/clulab/wm/eidos/groundings/grounders/SRLCompositionalGrounder.scala

```diff
@@ -162,11 +162,29 @@ class SRLCompositionalGrounder(name: String, domainOntology: DomainOntology, w2v
         val sortedSliced = groundings.sortBy(-_.score)
         sortedSliced.take(topN.getOrElse(sortedSliced.length))
     }
-
+    val returnedSRLGroundings = new ArrayBuffer[PredicateGrounding]
+    for (predGrounding <- srlGrounding) {
+      val tupleTheme = predGrounding.predicateTuple.theme
+      val tupleThemeProp = predGrounding.predicateTuple.themeProperties
+      val tupleProc = predGrounding.predicateTuple.themeProcess
+      val tupleProcProp = predGrounding.predicateTuple.themeProcessProperties
+      // move process property to theme property if process is empty AND no existing theme property
+      // THIS IS A HACK
+      if (tupleThemeProp.grounding.isEmpty && tupleProc.grounding.isEmpty && tupleProcProp.grounding.nonEmpty) {
+        val newTupleThemeProp = tupleProcProp
+        val newTupleProcProp = tupleProc
+        val newPredGrounding = PredicateGrounding(PredicateTuple(tupleTheme, newTupleThemeProp, tupleProc,
```

Comment:
`tupleTheme` here could be empty, right? is that desired (i.e., not checking)

Reply:
Good point...this would also give us the situation where we somehow have a `theme property` but no `theme`.